### DISCUSSION TITLE
Command idn as per mentioned in the pftl book is not working now

### DISCUSSION THREAD

**Discussion by BhargavRajGuru** [PFTL/pftl_discussions#2] [Q&A] [answered]
I am following the book precisely as I done before for my teachings. But after updating the arduino firmware from github repository, now device.readline() is not returning any value.

**Comment by aquilesC** [comment, marked as answer]
Hello! 
I'm sorry to hear it. The new firmware requires to use the following command: ``*IDN?`` (there were changes to the other commands as well). 
I am working on updating the contents of the book, in the meantime I would advise to use [this version of the firmware](https://github.com/PFTL/pythonforthelab/blob/458cd588e7d593dca15b58db1ab189e68d718c6b/extras/arduino_firmware/arduino_firmware.ino). Hope that solves the issues, keep me posted.

**Reply by BhargavRajGuru** [reply]
Okay, sure.

Thank you.
